Loop/Managers/LoopDataManager.swift

```diff
@@ -1482,6 +1510,18 @@ extension LoopDataManager {
     }
 }
 
+extension LoopDataManager {
+    public func updateTheLoop(completion: @escaping ([PredictedGlucoseValue]?, (recommendation: TempBasalRecommendation, date: Date)?, (recommendation: BolusRecommendation, date: Date)?) -> Void) {
+        self.dataAccessQueue.async {
+            do {
+                try self.update()
+                completion(self.predictedGlucose, self.recommendedTempBasal, self.recommendedBolus)
+            } catch let error {
+                self.logger.error("%{public}@", String(describing: error))
+            }
+        }
+    }
```

Comment:
I exposed the `update` function for testing purposes. I'm open to changing this, but the reason I chose to do it this way is so I could have a completion handler that would return the parts of the prediction/recommendations that were needed for testing.